Content.Shared/Mobs/Systems/MobThresholdSystem.cs

```diff
@@ -0,0 +1,330 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Content.Shared.Alert;
+using Content.Shared.Damage;
+using Content.Shared.FixedPoint;
+using Content.Shared.Mobs.Components;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Mobs.Systems;
+
+public sealed class MobThresholdSystem : EntitySystem
+{
+    [Dependency] private readonly MobStateSystem _mobStateSystem = default!;
+    [Dependency] private readonly AlertsSystem _alerts = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<MobThresholdsComponent, MapInitEvent>(MobThresholdMapInit);
+        SubscribeLocalEvent<MobThresholdsComponent, ComponentShutdown>(MobThresholdShutdown);
+        SubscribeLocalEvent<MobThresholdsComponent, DamageChangedEvent>(OnDamaged);
+        SubscribeLocalEvent<MobThresholdsComponent, ComponentGetState>(OnGetComponentState);
+        SubscribeLocalEvent<MobThresholdsComponent, ComponentHandleState>(OnHandleComponentState);
+        SubscribeLocalEvent<MobThresholdsComponent, UpdateMobStateEvent>(OnUpdateMobState);
+    }
+
+    private void OnHandleComponentState(EntityUid uid, MobThresholdsComponent component, ref ComponentHandleState args)
+    {
+        if (args.Current is not MobThresholdComponentState state)
+            return;
+
+        component.Thresholds = new SortedDictionary<FixedPoint2, MobState>(state.Thresholds);
+        component.CurrentThresholdState = state.CurrentThresholdState;
+    }
+
+    private void OnGetComponentState(EntityUid uid, MobThresholdsComponent component, ref ComponentGetState args)
+    {
+        args.State = new MobThresholdComponentState(component.CurrentThresholdState,
+            new Dictionary<FixedPoint2, MobState>(component.Thresholds));
+    }
+
+    private void MobThresholdMapInit(EntityUid uid, MobThresholdsComponent component, MapInitEvent args)
+    {
+        // TODO remove when body sim is implemented
+        EnsureComp<MobStateComponent>(uid);
+        EnsureComp<DamageableComponent>(uid);
+
+        if (!component.Thresholds.TryFirstOrNull(out var newState))
+            return;
+
+        component.CurrentThresholdState = newState.Value.Value;
+
+        TriggerThreshold(uid, MobState.Invalid, newState.Value.Value, thresholds: component);
+        UpdateAlerts(uid, newState.Value.Value, component);
+    }
+
+    private void MobThresholdShutdown(EntityUid uid, MobThresholdsComponent component, ComponentShutdown args)
+    {
+        if (component.TriggersAlerts)
+            _alerts.ClearAlertCategory(uid, AlertCategory.Health);
+    }
+
+    private void OnUpdateMobState(EntityUid uid, MobThresholdsComponent component, ref UpdateMobStateEvent args)
+    {
+        if (component.CurrentThresholdState != MobState.Invalid)
+            args.State = component.CurrentThresholdState;
+    }
+
+    public FixedPoint2 GetThresholdForState(EntityUid uid, MobState mobState,
```

Comment:
same for all public methods in this class. should maybe #region the private/public apis apart from eachother